crates/net/downloaders/src/headers/metrics.rs

```diff
@@ -0,0 +1,40 @@
+use metrics::Counter;
+use reth_interfaces::p2p::error::DownloadError;
+use reth_metrics_derive::Metrics;
+
+/// The header downloader metrics.
+#[derive(Metrics)]
+#[metrics(scope = "downloaders_headers")]
+pub struct HeaderDownloaderMetrics {
+    /// The number of headers that were successfully returned by the downloader.
+    total_returned: Counter,
+    /// Number of headers that were successfully downloaded
+    total_downloaded: Counter,
+    /// Number of timeout errors while requesting headers
+    timeout_errors: Counter,
+    /// Number of validation errors while requesting headers
+    validation_errors: Counter,
+    /// Number of unexpected errors while requesting headers
+    unexpected_errors: Counter,
+}
+
+impl HeaderDownloaderMetrics {
+    /// Increment the number of total downloaded headers.
+    pub(crate) fn update_total_returned(&self, number: usize) {
+        self.total_returned.increment(number as u64)
+    }
+
+    /// Increment the number of total downloaded headers.
+    pub(crate) fn update_total_downloaded(&self, number: usize) {
+        self.total_downloaded.increment(number as u64)
+    }
+
+    /// Update header errors metrics
+    pub(crate) fn update_error_metrics(&self, error: &DownloadError) {
```

Comment:
This one makes sense though
```suggestion
    pub(crate) fn increment_errors(&self, error: &DownloadError) {
```